lib/src/query_component.dart

```diff
@@ -31,16 +39,35 @@ class QueryComponent {
   final QueryService _queryService;
   final ExcelDart _excel;
 
+  QueryType _queryType;
+
   String advertiserId;
   String insertionOrderId;
-
   bool highlightUnderpacing = false;
 
   QueryComponent(this._queryService, this._excel);
 
+  bool get isAdvertiserQuery => _queryType == QueryType.byAdvertiser;
+  set isAdvertiserQuery(bool checked) =>
+      checked ? _queryType = QueryType.byAdvertiser : null;
+
+  bool get isInsertionOrderQuery => _queryType == QueryType.byInsertionOrder;
+  set isInsertionOrderQuery(bool checked) =>
+      checked ? _queryType = QueryType.byInsertionOrder : null;
+
   void onClick() async {
     // Uses DV360 public APIs to fetch entity data.
-    final insertionOrder = await _queryAndParseInsertionOrderEntityData();
+    var insertionOrders = await _queryAndParseInsertionOrderEntityData();
+
+    // If [_queryType.byAdvertiser], filters out insertion orders
+    // that are not in-flight.
+    insertionOrders = _queryType == QueryType.byAdvertiser
+        ? _filterInFlightInsertionOrders(insertionOrders)
+        : insertionOrders;
+
+    // TODO: update reporting query to deal with multiple IOs.
+    // Issue: https://github.com/googleinterns/dv360-excel-plugin/issues/61
+    final insertionOrder = insertionOrders[0];
```

Comment:
This is still querying the spend date for a single IO. Reporting api will be updated in the next pr.